### PR TITLE
feat: [SRM-10022]: Added custom time support for Logs

### PR DESCRIPTION
**Pull request by shashidhargg**
##### Summary:

Custom Calendar support in SLO Execution and External API Call Logs

##### Jira Links:

https://harness.atlassian.net/browse/SRM-10022

##### Screenshots:
Screenshot 2022-04-26 at 12 30 25 PM

Screenshot 2022-04-26 at 12 38 10 PM

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by shashidhargg** [comment]
retrigger jest

File: src/modules/85-cv/hooks/useLogContentHook/views/ExternalAPICall/ExternalAPICall.tsx [hunk 1]
```diff
@@ -40,9 +40,14 @@ const ExternalAPICall: React.FC<ExecutionAndAPICallLogProps> = props => {
     }
   }
 
-  const handleTimeRange = (_timeRange: SelectOption): void => {
-    setPageNumber(0)
-    setTimeRange?.(_timeRange)
+  const handleTimeRange = (newTimeRange: TimeRange): void => {
```
Comment (deepesh-ui) [inline src/modules/85-cv/hooks/useLogContentHook/views/ExternalAPICall/ExternalAPICall.tsx:43]: Similar function at  `src/modules/85-cv/hooks/useLogContentHook/views/ExecutionLog/ExecutionLog.tsx`.
can we move it to a common util  ?

File: src/modules/85-cv/hooks/useLogContentHook/views/LogContentHeader/CustomDatePicker/CustomDatePicker.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+                      e.stopPropagation()
+                      onChange(shortcut)
+                      setIsOpen(false)
+                    }}
+                  />
+                ))}
+              </Menu>
+            </Container>
+            <Container border={{ left: true }}>
+              <DatePicker value={date} maxDate={new Date()} onChange={dateValue => setDate(dateValue)} />
```
Comment (deepesh-ui) [inline src/modules/85-cv/hooks/useLogContentHook/views/LogContentHeader/CustomDatePicker/CustomDatePicker.tsx:59]:
```suggestion
              <DatePicker value={date} maxDate={new Date()} onChange={setDate} />
```

Comment (shashidhargg) [reply src/modules/85-cv/hooks/useLogContentHook/views/LogContentHeader/CustomDatePicker/CustomDatePicker.tsx:59]: Here `onChange` signature is like this - `(selectedDate: Date, isUserChange: boolean) => void`. So we can't directly pass the setDate.

Comment (shashidhargg) [reply src/modules/85-cv/hooks/useLogContentHook/views/LogContentHeader/CustomDatePicker/CustomDatePicker.tsx:59]: Moved the time range change check logic to utils. Please check.

File: src/modules/85-cv/hooks/useLogContentHook/views/ExternalAPICall/ExternalAPICall.tsx (hunk 1 above)
Comment (shashidhargg) [reply src/modules/85-cv/hooks/useLogContentHook/views/ExternalAPICall/ExternalAPICall.tsx:44]: Moved the time range change check logic to utils. Please check.

**Review by deepesh-ui — APPROVED** [review]

**Comment by shashidhargg** [comment]
trigger cypress

**Review by vasuki-kalluri — APPROVED** [review]

**Pull request merged**